test/commands/exec.test.ts

```diff
@@ -40,7 +40,21 @@ describe('exec', () => {
       expect(ctx.stdout).to.equal('worked\n');
     });
 
-   test
+  defaults
+    .stdout()
+    .command(['exec', '-a', account.name, '-e', environment.name, '-r', 'app:0', '--', 'ls', '-la'])
+    .it('exec component and replica of the form <service-name>:<replica-index>', ctx => {
+      expect(ctx.stdout).to.equal('worked\n');
+    });
+  
+  defaults
+    .stdout()
+    .command(['exec', '-a', account.name, '-e', environment.name, '-r', '0', '--', 'ls', '-la'])
+    .it('exec component and replica of the form <replica-index> when there is only one service', ctx => {
+      expect(ctx.stdout).to.equal('worked\n');
+    });
+
```

Comment:
We should throw a couple negative tests in here too that make sure the invalid inputs give us errors (like `-r -1` or `-r foo:`)